src/main/java/de/retest/recheck/RecheckImpl.java

```diff
@@ -33,6 +37,9 @@
 
 	private TestReplayResult currentTestResult;
 
+	private final Map<String, DefaultValueFinder> usedFinders = new HashMap<>();
```

Comment:
This somewhat implies that an instance of `RecheckImpl` may hold multiple `DefaultValueFinder`s and/or it may change at runtime. Can this be? Don't we have a contract?

Reply:
We are collecting all `DefaultValueFinder`s of each `check` to use them later in `capTest` to provide detailed information about the differences with respect to the corresponding default values. Thus it will change during a test method.

Reply:
Well, `RecheckImpl` gets the finder via `RecheckAdapter#getDefaultValueFinder()`, which is not context-sensitive. Can't we assume that the type, respectively, behavior of the finder is always the same, but only the instance may changes? I just think the implementation adds more complexity than needed at the moment.